### ISSUE TITLE
Feature request: static checks dependencies

### ISSUE THREAD

**Issue by ewjoachim** [floatingpurr/sync_with_poetry#34]
When using pre-commit with a static type checker (mypy, pyright, ...), one often needs to add dependencies to the hook so that the type checker knows how to navigate the project ([example](https://github.com/procrastinate-org/procrastinate/blob/main/.pre-commit-config.yaml#L31-L48))

It would be nice if sync_with_poetry would also sync those. For example, we could bind a hook name with a poetry group name in the `sync_with_poetry` args, and it would get all of the deps defined in the poetry group, put them in the expected format and write them back in the yaml ?

**Comment by ewjoachim** [comment]
Note: on some of my projects, I've made a script that does this, but I haven't integrated it as a pre-commit hook by itself: https://github.com/procrastinate-org/procrastinate/blob/main/scripts/pre-commit-pyright
One of the reasons is that it's hard to extract raw data from poetry files, so I'm afraid my script is brittle because it relies on parsing stdout outputs designed for humans.

I think it would feel less over-engineered if it was a part of this hook (or another hook altogether, but the concept is so close that it would require maintaining almost the same project twice)
(it could also be a different hook, but maintained in this repo)

**Comment by floatingpurr** [comment]
I see. Thanks for sharing this suggestion. Currently I have no time to analyze this use case in depth. I'll leave this issue open for future improvements or PRs.

**Comment by ewjoachim** [comment]
So hm, I wanted to see how hard it would be to work on that. I realized that implementing this would be almost as complex than the original hook. And then looked at how this hook is implemented (I had forgotten that we once had [a discussion](https://github.com/floatingpurr/sync_with_poetry/issues/8#issuecomment-1141324567) on that), so I tried doing it the way I had in mind, and... Pretty much ended up with my own pair of hooks (which uses ruamel's roundtrip and uses `poetry` itself for parsing poetry files).

I added your name in the credits and the license, as your project clearly is a big inspiration :)
